hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/ProcedureExecutor.java

```diff
@@ -2070,6 +2070,9 @@ private void execCompletionCleanup(Procedure<TEnvironment> proc) {
       // Catch NullPointerExceptions or similar errors...
       LOG.error("CODE-BUG: uncatched runtime exception for procedure: " + proc, e);
     }
+    // call schedulers completion cleanup, we have some special clean up logic in this method so if
+    // it throws any exceptions, we can not just ignore it, like the above procedure's cleanup
+    scheduler.completionCleanup(proc);
```

Comment:
```java
try {
  scheduler.completionCleanup(proc);
} catch (Throwable e) {
   // Catch NullPointerExceptions or similar errors...
   LOG.error("CODE-BUG: uncatched runtime exception for completion cleanup: {}", proc, e);
}
```
The code comments indicate that executing scheduler.completionCleanup(proc) may throw exception,
Don't we need to deal with it in here ?

Reply:
I think the intention for the old code is to catch the code bug and let the procedure executor go on.

Since now the schelder.completionCleanup has important logic, if there are errors we'd better throw them out and crash master, otherwise we may get some hanging procedures for ever...

Reply:
Understood, thank you for your reply.

Reply:
I'm not 100% sure that this change will crash the master. I think there are other protections that log-and-swallow exceptions that happen during procedure execution. But yeah, I agree that the master should be less optimistic about being in a healthy state when procedures fail. Let's see how this behaves, maybe a chaos monkey will find issue.